irescue/count.py

```diff
@@ -100,111 +100,126 @@ def compute_cell_counts(equivalence_classes, features_index, max_iters,
+                # find paths in list of nodes starting from parent
+                path = []
+                subg_copy = subg.copy()
+                nodes = [parent] + [x for x in subg_copy if x != parent]
+                for node in nodes:
+                    # make a copy of subgraph and remove nodes already used
+                    # in a path
+                    if node not in blacklist:
+                        path = pathfinder(subg_copy, node, path=[], features=None)
+                        for x in path:
+                            blacklist.append(x)
+                            subg_copy.remove_node(x)
+                        paths[parent].append(path)
+            # find the path configuration leading to the minimum number of
+            # deduplicated UMIs -> list of lists of nodes
+            path_config = [
+                paths[k] for k, v in paths.items()
+                if len(v) == min([len(x) for x in paths.values()])
+            ][0]
+            if not features:
+                # take features from parent node of selected path configuration
+                features = [list(subg.nodes[x[0]]['ft']) for x in path_config]
+            else:
+                # if features was already determined (i.e. no parent nodes),
+                # multiplicate the feature's list by the number of paths
+                # in path_config to avoid going out of list range
+                features *= len(path_config)
+            # assign UMI count to features
+            for feats in features:
+                if len(feats) == 1:
+                    counts[feats[0]] += 1.0
+                elif len(feats) > 1:
+                    row = [1 if x in feats else 0
+                        for x in range(1, number_of_features+1)]
+                    em_array.append(row)
+                else:
+                    writerr(nx.to_dict_of_lists(subg))
+                    writerr([subg.nodes[x]['ft'] for x in subg.nodes])
+                    writerr([subg.nodes[x]['c'] for x in subg.nodes])
+                    writerr(path_config)
+                    writerr(path)
+                    writerr(features)
+                    writerr(feats)
+                    writerr("Error: no common features detected in subgraph's"
+                            " path.", error=True)
+            # add EC log to dump
+            if dumpEC:
+                for i, path_ in enumerate(path_config):
+                    # add empty fields to parent node
+                    parent_ = path_[0]
+                    path_.pop(0)
+                    dump[parent_] += (b'', b'')
+                    # if child nodes are present, add parent node informations
+                    for x in path_:
+                        # add parent's UMI sequence and dedup features
+                        dump[x] += (dump[parent_][0], features[i])
+
+    else: ## in case of UMI-less
+        for x in equivalence_classes:
+            feats = list(x.features)
```

Comment:
Avoid using ambiguous variables ("x" is already used inside the loop)
```suggestion
        for eqc in equivalence_classes:
            feats = list(eqc.features)
```